api/src/main/java/net/kyori/adventure/translate/TranslationRegistry.java

```diff
@@ -0,0 +1,149 @@
+   *   registry.register("myplugin.hello", Locale.US, new MessageFormat("Hi, {0}. How are you?"));
+   * </pre>
+   *
+   * @param key a translation key
+   * @param locale a locale
+   * @param format a translation format
+   * @throws IllegalArgumentException if the translation key is already exists
+   */
+  void register(final @NonNull String key, final @NonNull Locale locale, final @NonNull MessageFormat format);
+
+  /**
+   * Registers a map of translations.
+   *
+   * @param locale a locale
+   * @param formats a map of translation keys to formats
+   * @throws IllegalArgumentException if a translation key is already exists
+   * @see #register(String, Locale, MessageFormat)
+   */
+  default void registerAll(final @NonNull Locale locale, final @NonNull Map<String, MessageFormat> formats) {
+    final List<IllegalArgumentException> errors = new LinkedList<>();
+    for(final Map.Entry<String, MessageFormat> entry : formats.entrySet()) {
+      try {
+        this.register(entry.getKey(), locale, entry.getValue());
+      } catch(final IllegalArgumentException e) {
+        errors.add(e);
+      }
+    }
+    final int size = errors.size();
+    if(size == 1) {
+      throw errors.get(0);
+    } else if(size > 1) {
+      throw new IllegalArgumentException(String.format("Invalid key (and %d more)", size - 1), errors.get(0));
+    }
+  }
+
+  /**
+   * Registers a resource bundle of translations.
+   *
+   * @param locale a locale
+   * @param resourceBundle a resource bundle
+   * @param escapeSingleQuotes whether to escape single quotes
+   * @throws IllegalArgumentException if a translation key is already exists
+   */
+  default void registerAll(final @NonNull Locale locale, final @NonNull ResourceBundle resourceBundle, final boolean escapeSingleQuotes) {
+    final List<IllegalArgumentException> errors = new LinkedList<>();
+    for(final String key : resourceBundle.keySet()) {
+      String format = resourceBundle.getString(key);
+      if(escapeSingleQuotes) {
+        format = SINGLE_QUOTE_PATTERN.matcher(format).replaceAll("''");
+      }
+      try {
+        this.register(key, locale, new MessageFormat(format, locale));
+      } catch(final IllegalArgumentException e) {
+        errors.add(e);
+      }
+    }
+    final int size = errors.size();
+    if(size == 1) {
+      throw errors.get(0);
+    } else if(size > 1) {
+      throw new IllegalArgumentException(String.format("Invalid key (and %d more)", size - 1), errors.get(0));
+    }
+  }
+
+  /**
+   * Registers a resource bundle of translations.
+   *
+   * @param locale a locale
+   * @param resourceBundlePath a resource bundle path
+   * @param escapeSingleQuotes whether to escape single quotes
+   * @throws IllegalArgumentException if a translation key is already exists
+   * @see #registerAll(Locale, ResourceBundle, boolean)
+   */
+  default void registerAll(final @NonNull Locale locale, final @NonNull String resourceBundlePath, final boolean escapeSingleQuotes) {
```

Comment:
Vanilla uses json files for translations, it would be interesting to have a resource bundle loader for this as well.